zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```diff
@@ -13,3 +18,120 @@ case class DataSetInfo(name: String,
+    }
+  }
+
+  //Needed for HierarchyField levels
+  implicit def tuple2Reads[A, B](implicit aReads: Reads[A], bReads: Reads[B]): Reads[Tuple2[A, B]] = Reads[Tuple2[A, B]] {
+    case JsArray(arr) if arr.size == 2 => for {
+      a <- aReads.reads(arr(0))
+      b <- bReads.reads(arr(1))
+    } yield (a, b)
+    case _ => JsError("Expected array of two elements")
+  }
+
+  implicit def tuple2Writes[A: Writes, B: Writes] = Writes[(A, B)](t => Json.obj("something1" -> t._1, "something1" -> t._2))
+
+  implicit val fieldFormat: Format[Field] = new Format[Field] {
+    override def reads(json: JsValue): JsResult[Field] = {
+      val name = (json \ "name").as[String]
+      val isOptional = (json \ "isOptional").as[Boolean]
+      DataType.withName((json \ "datatype").as[String]) match {
+        case DataType.Number =>
+          JsSuccess(NumberField(name, isOptional))
+        case DataType.Record => ???
+          //TODO think about Record type later
+        case DataType.Point =>
+          JsSuccess(PointField(name, isOptional))
+        case DataType.Bag =>
+          val innerType = (json \ "innerType").as[String]
+          JsSuccess(BagField(name, DataType.withName(innerType), isOptional))
+        case DataType.Boolean =>
+          JsSuccess(BooleanField(name, isOptional))
+        case DataType.Hierarchy =>
+          val innerType = (json \ "innerType").as[String]
+          val levels = (json \ "levels").as[Seq[(String, String)]]
+          JsSuccess(HierarchyField(name, DataType.withName(innerType), levels))
+        case DataType.Text =>
+          JsSuccess(TextField(name, isOptional))
+        case DataType.String =>
+          JsSuccess(StringField(name, isOptional))
+        case DataType.Time =>
+          JsSuccess(TimeField(name, isOptional))
+        case unknown: DataType.Value => JsError(s"field datatype invalid: $unknown")
+      }
+    }
+
+    override def writes(field: Field): JsValue = {
+      val name = field.name
+      val isOptional = field.isOptional
+      val dataType = field.dataType.toString
+      field match {
+        case record: RecordField => JsNull
+        case bag: BagField => JsObject(List("name" -> JsString(name), "isOptional" -> JsBoolean(isOptional), "datatype" -> JsString(dataType), "innerType" -> JsString(bag.innerType.toString)))
+        case hierarchy: HierarchyField => JsObject(List("name" -> JsString(name), "isOptional" -> JsBoolean(isOptional), "datatype" -> JsString(dataType), "innerType" -> JsString(hierarchy.innerType.toString)))
+        case basicField: Field => JsObject(List("name" -> JsString(name), "isOptional" -> JsBoolean(isOptional), "datatype" -> JsString(dataType)))
+        }
+    }
+  }
+
+  implicit val datetimeFormat: Format[DateTime] = new Format[DateTime] {
+    override def reads(json: JsValue) = {
+      val formatter = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
```

Comment:
you can use builtin Joda time format like `Reads.jodaDateReads("yyyy-MM-dd'T'HH:mm:ss.SSSZ")`. or `Writes.jodaDateReads("yyyy-MM-dd'T'HH:mm:ss.SSSZ")`